README.md

```diff
@@ -0,0 +1,489 @@
+# sanctuary-def
+
+sanctuary-def is a run-time type system for JavaScript. It facilitates the
+definition of curried JavaScript functions which are explicit about the number
+of arguments to which they may be applied and the types of those arguments.
+
+It is conventional to import the package as `$`:
+
+```javascript
+const $ = require('sanctuary-def');
+```
+
+The next step is to apply `$.create` to a list of types (an environment):
+
+```javascript
+const def = $.create($.env);
+```
+
+`$.env` is a list of types. One could augment it with custom types
```

Comment:
After reading this the first time, it was not clear to me why one would want to augment `$.env` with custom types. In the following examples, it appeared that you were declaring custom types, such as `Maybe`, `$Pair`, `Rank`, and `Suit` and then just passing them directly as arguments to `def`.

After reading again carefully and taking a look at the source code, I think the way it actually works is that you have to define all your types beforehand and pass them to `$.create` as part of the environment. Any type that you pass to `def` which is not part of the environment will not work. Is that correct? If so, I think you should clarify this in the section about the `$.create` function and consider adding an explicit call to `$.create` in the later example code.